tests/test_server.py

```diff
@@ -813,6 +813,85 @@ async def test_server_logs_address(server_config, caplog, async_finalizer):
         log_contains(caplog, "protocol.rest", logging.INFO, f"Server listening on {address}:")
 
 
+class MockConnection:
+    """
+    Mock connection class to simulate an asyncpg connection.
+    This class includes a close method to mimic closing a database connection.
+    """
+
+    async def close(self, timeout: int) -> None:
+        return
+
+
+@pytest.mark.parametrize("db_wait_time", ["20", "0"])
+async def test_bootloader_db_wait(monkeypatch, tmpdir, caplog, db_wait_time: str) -> None:
+    """
+    Tests the Inmanta server bootloader's behavior with respect to waiting for the database to be ready before proceeding
+    with the startup, based on the 'db_wait_time' configuration.
+    """
+    state_dir: str = tmpdir.mkdir("state_dir").strpath
+    config.Config.set("database", "wait_time", db_wait_time)
+    config.Config.set("config", "state-dir", state_dir)
+
+    state = {"first_connect": True}
+
+    async def mock_asyncpg_connect(*args, **kwargs) -> MockConnection:
+        """
+        Mock function to replace asyncpg.connect.
+        Will raise an Exception on the first invocation.
+        """
+        if state["first_connect"]:
+            state["first_connect"] = False
+            raise Exception("Connection failure")
+        else:
+            return MockConnection()
+
+    async def mock_start(self) -> None:
+        """Mocks the call to self.restserver.start()."""
+        return
+
+    monkeypatch.setattr("inmanta.server.protocol.Server.start", mock_start)
+    monkeypatch.setattr("asyncpg.connect", mock_asyncpg_connect)
+    caplog.set_level(logging.INFO)
+    caplog.clear()
+    ibl: InmantaBootloader = InmantaBootloader()
+    start_task: asyncio.Task = asyncio.create_task(ibl.start())
+    await start_task
+
+    if db_wait_time != "0":
+        log_contains(caplog, "inmanta.server.bootloader", logging.INFO, "Waiting for database to be up.")
+        log_contains(caplog, "inmanta.server.bootloader", logging.INFO, "Successfully connected to the database.")
+    else:
+        # If db_wait_time is "0", the wait_for_db method is not called,
+        # hence "Successfully connected to the database." log message will not appear.
+        log_doesnt_contain(caplog, "inmanta.server.bootloader", logging.INFO, "Successfully connected to the database.")
+
+    log_contains(caplog, "inmanta.server.server", logging.INFO, "Starting server endpoint")
+
+    await ibl.stop(timeout=15)
+
+
+@pytest.mark.parametrize("db_wait_time", ["20", "0"])
```

Comment:
```suggestion
@pytest.mark.parametrize("db_wait_time", ["2", "0"])
```

I don't want to add 20s to the build time.

Reply:
Does this number really matter? the DB is already up at that point so it should connect on the first try and never wait. I feel if this assumption doesn't hold other test will fail everywhere. But I don't mind changing it

Reply:
right